src/Aspire.Hosting.Azure/ExistingAzureResourceExtensions.cs

```diff
@@ -74,7 +74,7 @@ public static IResourceBuilder<T> RunAsExisting<T>(this IResourceBuilder<T> buil
     /// <param name="nameParameter">The name of the existing resource.</param>
     /// <param name="resourceGroupParameter">The name of the existing resource group, or <see langword="null"/> to use the current resource group.</param>
     /// <returns>The resource builder with the existing resource annotation added.</returns>
-    public static IResourceBuilder<T> PublishAsExisting<T>(this IResourceBuilder<T> builder, IResourceBuilder<ParameterResource> nameParameter, IResourceBuilder<ParameterResource>? resourceGroupParameter = null)
+    public static IResourceBuilder<T> PublishAsExisting<T>(this IResourceBuilder<T> builder, IResourceBuilder<ParameterResource> nameParameter, IResourceBuilder<ParameterResource>? resourceGroupParameter)
```

Comment:
Do we want to make the same change for the `RunAsExisting` methods? If someone specifies a `Azure:ResourceGroup` in their user secrets (and not a Prefix), the resource group can be existing. So it is more likely to work if you don't specify a resource group.

Reply:
Given @mitchdenny's comment https://github.com/dotnet/aspire/pull/7733#issuecomment-2675562736, I think we should make the same change to RunAsExisting. It gives us room to take a single string to mean the whole resourceID in the future.

Reply:
Agreed